operations/security_and_authentication/Restrict_Access_to_NCN_Images_S3_Bucket.md

```diff
@@ -0,0 +1,687 @@
+
+Unless otherwise noted, the procedure should be run from ncn-m001 (not PIT).  
+
+The procedure was tested on a CSM 1.2.0 GA System. 
+
+## Test Connectivity Before Applying the ACL
+
+Save the following script to a file (e.g., con_test.sh). 
+
+```
+#!/bin/bash
+
+
+SNCNS="$(grep 'ncn-s.*\.nmn' /etc/hosts | awk '{print $NF;}' | xargs)"
+SCSNS_NMN="$(echo $SNCNS | xargs -n 1 | sed -e 's/$/.nmn/g')"
+SCSNS_HMN="$(echo $SNCNS | xargs -n 1 | sed -e 's/$/.hmn/g')"
+SCSNS_CMN="$(echo $SNCNS | xargs -n 1 | sed -e 's/$/.cmn/g')"
+
+RADOS_HTTP_PORT="8080"
+HAPROXY_HTTP_PORT="80"
+HAPROXY_HTTPSPORT="443"
+
+PASS="PASS"
+FAIL="FAIL"
+
+function rados_test
+{
+   NODES="$1"
+   MSG="$2"
+   TTYPE="$3"
+
+   echo "[i] $MSG"
+   for n in $NODES
+   do
+      echo -n "  RADOS $n: "
+      if [ "$TTYPE" == "CONN_FAIL" ]
+      then
+         curl -sI --connect-timeout 2 http://${n}:${RADOS_HTTP_PORT}/ &> /dev/null
+         rc=$?
+         rc_pass=28
+      else
+         curl -I --connect-timeout 2 http://${n}:${RADOS_HTTP_PORT}/ 2>/dev/null | grep -q "200 OK"
+         rc=$?
+         rc_pass=0
+      fi
+
+      if [ $rc -eq $rc_pass ]
+      then
+         echo $PASS
+      else
+         echo $FAIL
+      fi
+   done
+}
+
+function haproxy_test
+{
+   NODES="$1"
+   MSG="$2"
+
+   echo "[i] $MSG"
+   for n in $NODES
+   do
+      echo -n "  HAPROXY (CEPH) HTTP $n: "
+      curl -I --connect-timeout 2 http://${n}:${HAPROXY_HTTP_PORT}/ncn-images/ 2>/dev/null | grep -q "x-amz-request-id"
+      if [ $? -eq 0 ]
+      then
+         echo $PASS
+      else
+         echo $FAIL
+      fi
+
+      echo -n "  HAPROXY (CEPH) HTTPS $n: "
+      curl -kI --connect-timeout 2 https://${n}:${HAPROXY_HTTPS_PORT}/ncn-images/ 2>/dev/null | grep -q "x-amz-request-id"
+      if [ $? -eq 0 ]
+      then
+         echo $PASS
+      else
+         echo $FAIL
+      fi
+
+   done
+}
+
+rados_test "$SCSNS_NMN" "MGMT RADOS over NMN"
+rados_test "$SCSNS_HMN" "MGMT RADOS over HMN"
+rados_test "$SCSNS_CMN" "MGMT RADOS over CMN" "CONN_FAIL"
+haproxy_test "$SCSNS_NMN" "MGMT HAProxy over NMN"
+```
+
+Execute the script, if the ACLs have not been applied, you should see results similar to those below. 
+
+```
+ncn-m001# bash ./con_test.sh 
+[i] MGMT RADOS over NMN
+  RADOS ncn-s001.nmn: PASS
+  RADOS ncn-s002.nmn: PASS
+  RADOS ncn-s003.nmn: PASS
+[i] MGMT RADOS over HMN
+  RADOS ncn-s001.hmn: PASS
+  RADOS ncn-s002.hmn: PASS
+  RADOS ncn-s003.hmn: PASS
+[i] MGMT RADOS over CMN
+  RADOS ncn-s001.cmn: FAIL
+  RADOS ncn-s002.cmn: FAIL
+  RADOS ncn-s003.cmn: FAIL
+[i] MGMT HAProxy over NMN
+  HAPROXY (CEPH) HTTP ncn-s001.nmn: PASS
+  HAPROXY (CEPH) HTTPS ncn-s001.nmn: PASS
+  HAPROXY (CEPH) HTTP ncn-s002.nmn: PASS
+  HAPROXY (CEPH) HTTPS ncn-s002.nmn: PASS
+  HAPROXY (CEPH) HTTP ncn-s003.nmn: PASS
+  HAPROXY (CEPH) HTTPS ncn-s003.nmn: PASS
+```
+
+## Configure HAProxy ACLs and Logging
+
+Necessary to limit RGW VIP access, to the ncn-images bucket, to management NCNs. 
```

Comment:
```suggestion
Necessary to limit RGW VIP access to the `ncn-images` bucket and to the management NCNs. 
```

Reply:
Access to the bucket is limited to management NCNs